source/features/more-dropdown.tsx

```diff
@@ -53,18 +59,40 @@ async function init(): Promise<void> {
 	const compareUrl = `/${repoUrl}/compare/${reference}`;
 	const commitsUrl = `/${repoUrl}/commits/${reference}`;
 	const dependenciesUrl = `/${repoUrl}/network/dependencies`;
+	const branchesUrl = `/${repoUrl}/branches`;
 
 	const nav = select('.js-responsive-underlinenav .UnderlineNav-body');
+
 	if (nav) {
 		// "Repository refresh" layout
+		const menu = select('.js-responsive-underlinenav-overflow ul')!;
+
 		nav.parentElement!.classList.add('rgh-has-more-dropdown');
-		select('.js-responsive-underlinenav-overflow ul')!.append(
+		menu.append(
 			<li className="dropdown-divider" role="separator"/>,
-			createDropdownItem('Compare', compareUrl),
-			pageDetect.isEnterprise() ? '' : createDropdownItem('Dependencies', dependenciesUrl),
-			createDropdownItem('Commits', commitsUrl),
-			createDropdownItem('Branches', `/${repoUrl}/branches`)
+			createDropdownItem('Compare', compareUrl, {}, DiffIcon),
+			pageDetect.isEnterprise() ? '' : createDropdownItem('Dependencies', dependenciesUrl, {}, PackageIcon),
+			createDropdownItem('Commits', commitsUrl, {}, HistoryIcon),
+			createDropdownItem('Branches', branchesUrl, {}, BranchIcon)
 		);
+
+		for (const tab of select.all<HTMLAnchorElement>('.js-responsive-underlinenav-item')) {
+			const shouldBeRemoved = uselessTabs.has(tab.dataset.tabItem!);
+
+			if (shouldBeRemoved) {
+				tab.parentElement!.remove();
+			}
+
+			menu.replaceChild(
+				<li data-menu-item={tab.dataset.tabItem} hidden={!shouldBeRemoved && tab.style.visibility !== 'hidden'}>
+					<a href={tab.href} className="rgh-reponav-more dropdown-item">
+						{[...tab.cloneNode(true).childNodes]}
+					</a>
+				</li>,
+				menu.querySelector(`[data-menu-item="${tab.dataset.tabItem!}"]`)!
+			);
```

Comment:
You can use `ChildNode.replaceWith` instead of the old `ParentNode.replaceChild`
```suggestion
			menu.querySelector(`[data-menu-item="${tab.dataset.tabItem!}"]`)!.replaceWith(
				<li data-menu-item={tab.dataset.tabItem} hidden={!shouldBeRemoved && tab.style.visibility !== 'hidden'}>
					<a href={tab.href} className="rgh-reponav-more dropdown-item">
						{[...tab.cloneNode(true).childNodes]}
					</a>
				</li>
			);
```

Reply:
But, what is this doing? Why are replacing _every_ link?

Reply:
> Why is it replacing _every_ link?

because of the icons
![Screenshot from 2020-08-29 02-11-11](https://user-images.githubusercontent.com/36174850/91622700-146e4980-e9a1-11ea-9dd2-ba1591b5cbe7.png)

![Screenshot from 2020-08-29 02-17-37](https://user-images.githubusercontent.com/36174850/91622703-16d0a380-e9a1-11ea-815c-6ebe19917827.png)

Reply:
Can it specifically add the icons instead of having to recreate each item from scratch?

Reply:
> having to recreate each item from scratch?

We can recreate only `style="visibility: hidden;"` items